api/v1beta1/openstackdataplanerole_types.go

```diff
@@ -66,6 +66,11 @@ type OpenStackDataPlaneRoleSpec struct {
 	// +kubebuilder:default={configure-network,validate-network,install-os,configure-os,run-os}
 	// Services list
 	Services []string `json:"services"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="quay.io/openstack-k8s-operators/openstack-ansibleee-runner:latest"
```

Comment:
Well it's already  defaulted in AnsibleEESpec https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/api/v1beta1/common.go#L157-L159.

I can see the webhook doing it in https://github.com/openstack-k8s-operators/openstack-ansibleee-operator/blob/main/api/v1alpha1/openstack_ansibleee_types.go#LL201C37-L201C64. We can remove from both the places. I'll update the patch.

Reply:
Thinking again, the use-case I was looking for is to override the openstack-ansibleee-runner image in CI after building with ansible-edpm PR and felt that doing it for all services is not efficient. Now, I think we can do it without any changes here by providing the ANSIBLEEE_IMAGE_URL_DEFAULT ENV variable to `make openstack`. 

I guess the only reason we want to have  OpenStackAnsibleEERunnerImage at the service level is for users to use a different runner image for their 'custom` services?  I feel setting something for a 'role' and then overriding it with service (which is deployment specific) would be an anti-pattern.

We can probably avoid all the hack for passing the image with OPENSTACK_RUNNER_IMG/DATAPLANE_RUNNER_IMG and just use ANSIBLEEE_IMAGE_URL_DEFAULT in edpm-ansible jobs after we build the image with epdm-ansible PR.

Reply:
> Thinking again, the use-case I was looking for is to override the openstack-ansibleee-runner image in CI after building with ansible-edpm PR and felt that doing it for all services is not efficient. Now, I think we can do it without any changes here by providing the ANSIBLEEE_IMAGE_URL_DEFAULT ENV variable to `make openstack`.
> 
> I guess the only reason we want to have OpenStackAnsibleEERunnerImage at the service level is for users to use a different runner image for their 'custom` services? I feel setting something for a 'role' and then overriding it with service (which is deployment specific) would be an anti-pattern.

A custom image with custom ansible content is the primary reason, yes. Not sure it's the only reason. Could you elaborate on what you would consider the anti-pattern? 

> We can probably avoid all the hack for passing the image with OPENSTACK_RUNNER_IMG/DATAPLANE_RUNNER_IMG and just use ANSIBLEEE_IMAGE_URL_DEFAULT in edpm-ansible jobs after we build the image with epdm-ansible PR.

Yes, i think that would work as well for CI.

Reply:
> Could you elaborate on what you would consider the anti-pattern?

datasplaneervice definitions are unique in a namspace (for an overcloud deployment/dataplane). Role is kinda subset of dataplane. If override runner image  for a role with runner image for service(global) it would be an anti-pattern IMO. We can't do it the other way around too as service is more granular where you can override the image per service.